### ISSUE TITLE
legacy Java SE 6 runtime

### ISSUE THREAD

**Issue by GuyBaele** [beast-dev/tracer#75]
When downloading the latest Tracer version for Mac from Andrew's website (Tracer v1.6.0) and trying to run it, Sierra complains about the need to install the legacy Java SE 6 runtime, despite the fact that I have JRE 8 update 121 running. It would be awesome if this could be sorted out for the new release.

**Comment by walterxie** [comment]
@rambaut @alexeid 

Hi, Andrew and Alexei,

This is easy to solve if we move to Java 8. Are we ready yet? Most Mac users are using Sierra now. It'd better to upgrade BEAST 1 to the same Java version if Tracer is upgrading.

In addition, would this release be a major release 1.7.0 if many big changes? 

Cheers,
Walter

**Comment by walterxie** [comment]
Alexei just mentioned we could simply compile both project on Java 8 without any change.

**Comment by rambaut** [comment]
Compiling to Java 8 and running on JVM 8 are two different things. The issue is not what the code is compiled with but the script within the Mac package that works out what JVM is installed and runs it. It doesn't pick the JVM version based on the language version of the JAR file.

On the Mac it is not sufficient to have the JRE 1.8 installed as that just has provides a browser plug-in. To run BEAST or Tracer or any other Java desktop program it is necessary to install the JDK. 

The Apple Java 1.6 install still works fine and is easier than the JDK 1.8 install. Probably the best solution (and one I will look into for FigTree also) is including a JVM within the Mac application package.

So - I would absolutely be against compiling the code to Java 8 language unless there was an absolutely compelling case that the language features it brings are worth it (possibly for a newly written package but not for old code, I suspect). I see no point in making the JAR incompatible with older JVMs just for the sake of it. 

A.

**Comment by GuyBaele** [comment]
Just to be clear, I do have the JDK 1.8 installed as well, which prompted my question as to why I would have to install a 1.6 JDK as well.

**Comment by rambaut** [comment]
This is an issue with the run script in the Mac app package. Will investigate.

**Comment by rvosa** [comment]
I came across this issue by way of Google, so others might get here as well. Today I downloaded the Tracer dmg and tried opening the app on osx 10.12.6 with java 8 as the default. I also got the error telling me to downgrade, so the app bundler still runs the script tailored for java 6. However, the command line (linux/osx) version runs fine under java 8. So your better option is to just run that, I assume an updated launcher script is in the works.
